generator/src/main/java/com/scottlogic/deg/generator/decisiontree/ConstraintNode.java

```diff
@@ -93,76 +73,15 @@ public String toString(){
                 : Objects.toString(atomicConstraints));
     }
 
-    public ConstraintNode removeDecisions(Collection<DecisionNode> decisionsToRemove) {
-        Function<DecisionNode, Boolean> shouldRemove = existingDecision -> decisionsToRemove.stream()
-            .anyMatch(decisionToExclude -> decisionToExclude.equals(existingDecision));
-
-        return new ConstraintNode(
-          this.atomicConstraints,
-          decisions.stream()
-              .filter(existingDecision -> !shouldRemove.apply(existingDecision))
-              .collect(Collectors.toList()),
-            this.nodeMarkings
-        );
+    public ConstraintNodeBuilder builder() {
+        return new ConstraintNodeBuilder(atomicConstraints,decisions,nodeMarkings);
```

Comment:
Normally I would add spaces between parameters, like `return new ConstraintNodeBuilder(atomicConstraints, decisions, nodeMarkings);`. Not a big deal though.